### PR TITLE
Property Where Syntax update

### PR DESCRIPTION
**Pull request by sidprasad**
Tests now take a form that more transparently shows the implicit implications.

```
assert p  is necessary for q for {...bounds... / scope}
assert p is sufficient for q for {...bounds... / scope}
```

where p and q are implications.

### CODE REVIEW

File: forge/tests/error/main.rkt [hunk 1]
```diff
@@ -80,8 +80,9 @@
     (list "failed_theorem.frg" #rx"failed.")
     (list "failed_sat.frg" #rx"Failed test")
     (list "failed_unsat.frg" #rx"Failed test")
-    (list "properties_undirected_tree_underconstraint_error.frg" #rx"isUndirectedTree implies TreeWithEdges failed. Found instance")
-    (list "properties_undirected_tree_overconstraint_error.frg" #rx"isUndirected implies isUndirectedTree failed. Found instance")
+    (list "failed_sat.frg" #rx"Failed test") 
+    (list "properties_undirected_tree_underconstraint_error.frg" #rx"(isUndirectedTree implies TreeWithEdges) failed. Found instance")
```
Comment (bennn) [inline forge/tests/error/main.rkt:84]: `#rx` is a regular expression, not a plain string. But what about a totally different error message? It's an `assert` now. `Assertion 'A is necessary for B' failed. Found instance` ?

cc @tnelson

Comment (sidprasad) [reply forge/tests/error/main.rkt]: I like this. I'm not sure how to intercept and change the error message though, since Forge thinks this is a theorem.

Comment (bennn) [reply forge/tests/error/main.rkt]: Oh, right. Then let's stick to `Theorem` for now.

Comment (bennn) [reply forge/tests/error/main.rkt]: You can always check the error tests locally by running `racket main.rkt`. To speed it up, comment out the other tests.

**Comment by bennn** [comment]
`Theorem Assertion` is weird, but ok we can always change!

**Pull request merged**